rpc/client/client.go

```diff
@@ -817,6 +829,25 @@ func (c *Client) CheckTx(ctx context.Context, tx tmtypes.Tx) (*ctypes.ResultChec
 	return &ctypes.ResultCheckTx{ResponseCheckTx: *res}, nil
 }
 
+// BlockValidated returns the settlement validation status for a specific height: 0: Not validated, 1: Validated from P2P, 2: Validated from settlement, -1: node is still syncing.
+func (c *Client) BlockValidated(ctx context.Context, height *int64) (*ResultBlockValidated, error) {
+	// invalid height
+	if *height < 0 || height == nil {
```

Comment:
this can panic. for short circuit evaluation order is important, the OR is evaluated from left to right

```suggestion
	if height == nil || *height < 0 {
```

Reply:
fixed